src/plugins.ts

```diff
@@ -0,0 +1,16 @@
+type BuilderFacade = {
+  // addType: (def: any) => void;
+  hasType: (typeName: string) => boolean;
+};
+
+type Plugin = (hooks: Hooks) => void;
+
+type Hooks = {
+  onInstall: OnInstallHook;
+};
+
+type OnInstallHook = (
+  onInstallHandler: (builder: BuilderFacade) => { types: any[] }
```

Comment:
I'm proposing here that the plugin hook handler uses `return` values to "write" back to nexus core, and keeps the passed builder facade argument for read-only operations. I think this lends to a data flow that is easier to understand. The plugin builds up types however it wants and _returns_ them to nexus core. Seems natural this way.

Reply:
With #242 in sight, I retract my proposition : )

Reply:
We agreed about this on slack chat.